enhancements/network/external-lb-vips.md

```diff
@@ -0,0 +1,214 @@
+  remain the HAproxy/Keepalived/CoreDNS stack.
+* Supporting a mix of both (internal + external loadbalancers for the API & Ingress VIPs), customers
+  will have to choose a scenario.
+
+## Proposal
+
+We will make some change in `install-config.yaml` so customer can configure the
+deployment to use an external loadbalancer if they want.
+Then the components which interact with networking will have to check this
+new field and handle this scenario appropriately.
+
+### Workflow Description
+
+1. The customer will configure their external loadbalancer(s), which includes configuring the VIPs
+   for API and Ingress and also configure the loadbalancer tool that is being used (e.g. HAproxy, F5, etc)
+2. The customer will write the `install-config.yaml` file and set the new `loadBalancerType` platform specific field to `External`, and
+   will set the fields for `apiVIP` and `ingressVIP`.
+3. In the installer, if the platform does support external loadbalancer, some validation will be run
+   to check that the VIPs can already be resolved (the validation already exists for vsphere, we need to
+    make it re-usable). The new field for `LoadBalancerType` will be set in `PlatformSpec`.
+4. In machine-config-operator, if LB type is external, the internal loadbalancer will not be
+   deployed.
+
+### API Extensions
+
+A new field will be added to the `PlatformStatus` of the infrastructure object to tell
+the kind of the loadbalancer.
+
+```golang
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+// +kubebuilder:validation:Enum:="Internal";"External"
+type PlatformLoadBalancerType string
+const (
+  // LoadBalancerTypeExternal is a loadbalancer that is managed outside of the cluster by the customer.
+  LoadBalancerTypeExternal LoadBalancerType = "CustomerManaged"
+
+  // LoadBalancerTypeInternal is a loadbalancer that is managed by the OpenShift cluster.
+  LoadBalancerTypeInternal LoadBalancerType = "OpenShiftManaged"
+)
+// VSpherePlatformSpec holds the desired state of the vSphere infrastructure provider.
+type VSpherePlatformSpec struct {
+  // loadBalancerType defines if an internal or external load balancer
+  // is to be used for the openshift api and ingress endpoints.
+  // When set to Internal the static pods in charge of API and Ingress traffic load-balancing
+  // defined in the machine config operator will be deployed.
+  // When set to External these static pods will not be deployed and it is expected that
+  // the load balancer is configured out of band by the deployer.
+  // When omitted, this means no opinion and the platform is left to choose a reasonable default
+  // The current default value is Internal.
+  // +optional
+  LoadBalancerType PlatformLoadBalancerType `json:"loadBalancerType,omitempty"`
+
+}
+```
+
+This field will be set in the `install-config.yaml`:
+```yaml
+platform:
+  vsphere:
+    loadBalancerType: CustomerManaged
```

Comment:
here I'm wondering if it'll be better UX to provide a boolean rather than expecting the customer to write `CustomerManaged`. Something like `externaLB: true` or something. And in the installer we set the proper fields in `PlatformSpec`.
WDYT? @patrickdillon @JoelSpeed

Reply:
I think it's a good point in terms of UX, but in general enums are preferred in the API. Installer validation should catch any typos, so the ux experience should not be so bad.

Reply:
I think `CustomerManaged` may have been my suggestion, so this may be my fault: having "Customer" in the value is probably not great. 

"UserManaged" would be better and perhaps sufficient; the point I was trying to make was that I wasn't sure if the API field accurately (and clearly) reflected the solution.

Reply:
ack, I'll reflect that change. Thanks for this feedback.

Reply:
I know this is resolved, but, just coming to this, IMO, we want a consistent API between the install and running cluster where possible, so the install config should use an enum too